arduino/smartcar/smartcar.ino

```diff
@@ -155,7 +164,68 @@ void loop()
     
 }
 
-//If the connection breaks, this method will be called
+
+void stopZoneAutoBreak(long frontUltDis, long frontIRDis, long backIRDis)
+{
+     if (frontUltDis <= 100 && frontUltDis != 0 || frontIRDis <= 40 && frontIRDis != 0 || backIRDis <= 40 && backIRDis != 0)//stop zone
+         {
+            if (canDrive)//check whether you're in the stop zone
+            {
+                car.setSpeed(0);
+                Serial.println("Emergency stop 1");
+            }
+            canDrive = false;//so the car can move in the stop soon
+        } else {
+            canDrive = true;//so the car will stop again if it hits the stop zone
+        }
+}
+
+void incomingAvoidanceThreshold(long frontUltDis, long frontIRDis, long backIRDis)
+{
+    if (frontUltDis <= 30 && frontUltDis != 0 || frontIRDis <= 40 && frontIRDis != 0)//Ford threshold 1
+    {
+        car.setSpeed(0);
+        car.setSpeed(-90);
+        Serial.println("backing up level 1");
+        activeAvoidance = true;
+    } else if (frontUltDis <= 60 && frontUltDis != 0 || frontIRDis <= 40 && frontIRDis != 0)//Ford threshold 2
+    {
+        car.setSpeed(0);
+        car.setSpeed(-60);
+        Serial.println("backing up level 2");
+        activeAvoidance = true;
+    } else if (frontUltDis <= 90 && frontUltDis != 0 || frontIRDis <= 40 && frontIRDis != 0)//Ford threshold 3this
+    {
+        car.setSpeed(0);
+        car.setSpeed(-30);
+        Serial.println("backing up level 3");
+        activeAvoidance = true;
+    } else if (backIRDis <= 10 && backIRDis != 0)//back threshold 1
+    {
+        car.setSpeed(0);
+        car.setSpeed(40);
+        Serial.println("moving forward level 1");
+        activeAvoidance = true;
+    } else if (backIRDis <= 20 && backIRDis != 0)//back threshold 2
+    {
+        car.setSpeed(0);
+        car.setSpeed(30);
+        Serial.println("moving forward level 2");
+        activeAvoidance = true;
+    } else if (backIRDis <= 30 && backIRDis != 0)//back threshold 3
+    {
+        car.setSpeed(0);
+        car.setSpeed(20);
+        Serial.println("moving forward level 3");
+        activeAvoidance = true;
+    } else if (frontUltDis == 0 && frontIRDis == 0 && backIRDis == 0 && activeAvoidance)
+    {
+        car.setSpeed(0);
+        activeAvoidance = false;
+    } 
+}
+
+//This method will be called when the connection breaks from the broker 
```

Comment:
I added my comment back here above my method

Reply:
Thank you I appreciate that